src/libexpr/attr-path.cc

```diff
@@ -68,10 +68,10 @@ Value * findAlongAttrPath(EvalState & state, const string & attrPath,
             if (attr.empty())
                 throw Error(format("empty attribute name in selection path '%1%'") % attrPath);
 
-            Bindings::iterator a = v->attrs->find(state.symbols.create(attr));
-            if (a == v->attrs->end())
+            Bindings::find_iterator a = v->attrs->find(state.symbols.create(attr));
```

Comment:
A lot of this code predates C++11, but if we're changing this let's just go to `auto` here.